### PR TITLE
Fix ConnectionUtil/ConnectionManager NPE

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 1]
```diff
@@ -35,35 +34,40 @@
     static final CurrentNetwork NO_NETWORK = new CurrentNetwork(NetworkState.NO_NETWORK_AVAILABLE, null);
     static final CurrentNetwork UNKNOWN_NETWORK = new CurrentNetwork(NetworkState.TRANSPORT_UNKNOWN, null);
 
-    private final ConnectionMonitor connectionMonitor;
+    private final Supplier<NetworkRequest> createNetworkMonitoringRequest;
     private final NetworkDetector networkDetector;
+    private final ConnectivityManager connectivityManager;
 
-    private final AtomicReference<CurrentNetwork> currentNetwork = new AtomicReference<>();
+    private volatile CurrentNetwork currentNetwork;
```
Comment (mateuszrzeszutek) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:41]: We don't use CAS anywhere, so I replaced an `AtomicReference` with a `volatile` field

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 2]
```diff
@@ -35,35 +34,40 @@
...
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            connectivityManager.registerDefaultNetworkCallback(connectionMonitor);
-        } else {
-            NetworkRequest networkRequest = createNetworkMonitoringRequest.get();
-            connectivityManager.registerNetworkCallback(networkRequest, connectionMonitor);
-        }
+        this.createNetworkMonitoringRequest = createNetworkMonitoringRequest;
         this.networkDetector = networkDetector;
+        this.connectivityManager = connectivityManager;
         refreshNetworkStatus();
```
Comment (jkwatson) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:53]: I think this call should be moved down to `start`. See my implementation in #146  :)

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 3]
```diff
@@ -35,35 +34,40 @@
...
     }
 
     CurrentNetwork refreshNetworkStatus() {
         CurrentNetwork activeNetwork = networkDetector.detectCurrentNetwork();
-        currentNetwork.set(activeNetwork);
+        currentNetwork = activeNetwork;
         return activeNetwork;
     }
 
+    void start() {
```
Comment (jkwatson) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:62]: maybe rename to "startMonitoring"

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java (hunk 2 above)
Comment (jkwatson) [reply splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:53]: also note that you don't need to keep references to them if they just get passed into the `start` method with my approach.

File: splunk-otel-android/src/test/java/com/splunk/rum/ConnectionUtilTest.java [hunk 4]
```diff
@@ -115,4 +121,40 @@ public void quiznos() {
         monitorCaptor.getValue().onLost(null);
         assertEquals(2, notified.get());
     }
+
+    @Test
+    @Config(sdk = Build.VERSION_CODES.LOLLIPOP)
```
Comment (jkwatson) [inline splunk-otel-android/src/test/java/com/splunk/rum/ConnectionUtilTest.java:126]: I discovered that you can give this an array of `sdk` entries to collapse to a single test.

Comment (mateuszrzeszutek) [reply splunk-otel-android/src/test/java/com/splunk/rum/ConnectionUtilTest.java:126]: That won't work in this case unfortunately: https://github.com/signalfx/splunk-otel-android/pull/146#discussion_r704265207

**Review by jkwatson — APPROVED** [review]
Ship it!

**Pull request merged**